lib/user-interface/react/src/components/utils.ts

```diff
@@ -78,3 +78,44 @@ export const getDisplayableMessage = (content: MessageContent, ragCitations?: st
     }
     return content + (ragCitations ?? '');
 };
+
+export const fetchImage = async (url: string) => {
+    try {
+        const response = await fetch(url, {
+            method: 'GET',
+        });
+        if (!response.ok) {
+            throw new Error(`HTTP error! status: ${response.status}`);
+        }
+        return await response.blob();
+    } catch (error) {
+        console.error('Error fetching image:', error);
+        throw error;
+    }
+};
+
+export function messageContainsImage (content: MessageContent): boolean {
+    if (Array.isArray(content)) {
+        return !!content.find((item) => item.type === 'image_url');
+    }
+    return false;
+}
+
+export function base64ToBlob (base64: string, mimeType: string) {
+    const byteCharacters = atob(base64);
+    const byteArrays = [];
+
+    for (let i = 0; i < byteCharacters.length; i += 512) {
+        const slice = byteCharacters.slice(i, i + 512);
+        const byteNumbers = new Array(slice.length);
+
+        for (let j = 0; j < slice.length; j++) {
+            byteNumbers[j] = slice.charCodeAt(j);
+        }
+
+        const byteArray = new Uint8Array(byteNumbers);
+        byteArrays.push(byteArray);
+    }
+
+    return new Blob(byteArrays, { type: mimeType });
```

Comment:
Why are we doing this in 512 chunks? Since we have the whole base64 string from the start I think this can be simplified...

```typescript
function base64ToBlob(base64: string, mimeType = ''): Blob {
    const byteCharacters = atob(base64);
    const byteNumbers = new Array(byteCharacters.length);

    for (let i = 0; i < byteCharacters.length; i++) {
        byteNumbers[i] = byteCharacters.charCodeAt(i);
    }

    const byteArray = new Uint8Array(byteNumbers);
    return new Blob([byteArray], { type: mimeType });
}
```